service/dap/server.go

```diff
@@ -1470,18 +1501,33 @@ func (s *Server) onVariablesRequest(request *dap.VariablesRequest) {
 			children[i] = dap.Variable{
 				Name:               name,
 				EvaluateName:       cfqname,
+				Type:               s.getTypeIfSupported(c),
 				Value:              cvalue,
 				VariablesReference: cvarref,
 			}
 		}
 	}
+	if !s.clientCapabilities.supportsVariableType {
+		// If the client does not support variable type
+		// we cannot set the Type field in the response.
+		for i := range children {
+			children[i].Type = ""
+		}
+	}
 	response := &dap.VariablesResponse{
 		Response: *newResponse(request.Request),
 		Body:     dap.VariablesResponseBody{Variables: children},
 	}
 	s.send(response)
 }
 
+func (s *Server) getTypeIfSupported(v *proc.Variable) string {
+	if !s.clientCapabilities.supportsVariableType {
+		return ""
+	}
+	return v.TypeString()
```

Comment:
Would you mind changing the definition of TypeString in proc to be like this:

```
func (v *Variable) TypeString() string {
	if v == nilVariable {
		return "nil"
	}
	if v.DwarfType == nil {
		return v.Kind.String()
	}
	if v.DwarfType.Common().Name != "" {
		return v.DwarfType.Common().Name
	}
	r := v.DwarfType.String()
	if r == "*void" {
		if i := v.DwarfType.Common().Index; i < len(v.bi.Images) && len(v.bi.Images[i].compileUnits) > 0 && v.bi.Images[i].compileUnits[0].isgo {
			r = "unsafe.Pointer"
		}
	}
	return r
}
```

Reply:
PS. edited

Reply:
Done!